dotnet/src/SemanticKernel/AI/OpenAI/Clients/OpenAIClientAbstract.cs

```diff
@@ -132,14 +143,18 @@ protected virtual void Dispose(bool disposing)
     // HTTP user agent sent to remote endpoints
     private const string HTTPUseragent = "Microsoft Semantic Kernel";
 
-    private async Task<T> ExecutePostRequestAsync<T>(string url, string requestBody)
+    private async Task<T> ExecutePostRequestAsync<T>(string url, string requestBody, int httpTimeoutInSeconds)
     {
+        if (httpTimeoutInSeconds < 1) throw new AIException(AIException.ErrorCodes.UnknownError, "Invalid http timeout");
+        CancellationTokenSource cancellationSource = new CancellationTokenSource();
+        cancellationSource.CancelAfter(TimeSpan.FromSeconds(httpTimeoutInSeconds));
+
         string responseJson;
 
         try
         {
             using HttpContent content = new StringContent(requestBody, Encoding.UTF8, MediaTypeNames.Application.Json);
-            HttpResponseMessage response = await this.HTTPClient.PostAsync(url, content);
+            HttpResponseMessage response = await this.HTTPClient.PostAsync(url, content, cancellationSource.Token);
```

Comment:
I was going to plumb the CancellationTokenSource from the SKContext of the execution as well, so eventually this might look like:

```
        using var cancellationSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
        cancellationSource.CancelAfter(TimeSpan.FromSeconds(httpTimeoutInSeconds));
        ```

Reply:
This is mostly a note for myself or whoever ends up doing the merge resolution here :)

Reply:
If we use a cancellation token that is passed in, we can't really tell if `CancelAfter()` has been called on the source. This could cause accidental override or inifinte timeout on the http client.